cdb2api/cdb2api.c

```diff
@@ -4746,30 +4746,26 @@ static int cdb2_run_statement_typed_int(cdb2_hndl_tp *hndl, const char *sql,
     PRINT_AND_RETURN(-1);
 }
 
-static char *cdb2_type_str(int type)
-{
-    switch (type) {
-    case CDB2_INTEGER:
-        return "CDB2_INTEGER";
-    case CDB2_REAL:
-        return "CDB2_REAL";
-    case CDB2_CSTRING:
-        return "CDB2_CSTRING";
-    case CDB2_BLOB:
-        return "CDB2_BLOB";
-    case CDB2_DATETIME:
-        return "CDB2_DATETIME";
-    case CDB2_INTERVALYM:
-        return "CDB2_INTERVALYM";
-    case CDB2_INTERVALDS:
-        return "CDB2_INTERVALDS";
-    case CDB2_DATETIMEUS:
-        return "CDB2_DATETIMEUS";
-    case CDB2_INTERVALDSUS:
-        return "CDB2_INTERVALDSUS";
-    default:
+const char *cdb2_tp_str[] = {
+    "???",
+    "CDB2_INTEGER",
+    "CDB2_REAL",
+    "CDB2_CSTRING",
+    "CDB2_BLOB",
+    "???",
+    "CDB2_DATETIME",
+    "CDB2_INTERVALYM",
+    "CDB2_INTERVALDS",
+    "CDB2_DATETIMEUS",
+    "CDB2_INTERVALDSUS"
+};
+
+inline const char *cdb2_type_str(int type)
```

Comment:
Maybe use #2242 to get this instead of exposing another API call?

Reply:
Its ok. im leaving cdb2api out of this one, as discussed.